examples/fixnum.kl

```diff
@@ -0,0 +1,19 @@
+#lang "prelude.kl"
+
+
+
+(example (the Integer 0b1100))
+(example (the Word64 0xFF))
+(example (the Word32 0o32))
+(example (the Word16 12))
+(example (the Word8 12))
+(example (the Int64 12))
+(example (the Int32 12))
+(example (the Int16 12))
+(example (the Int8  256))  -- note no bounds checking with the primitives
+(example (word8-compliment (the Word8 7))) -- will be -8 because 2's complement
+(example (int32-popcount (the Int32 255) )) -- 8
+(example (int8-+ (the Int8 42) (the Int8 1726)))
```

Comment:
since the types are

```
word8-compliment : Word8 → Word8
int32-popcount : Int32 → Int32
int8-+ : Int8 → Int8 → Int8
```

I would have expected the type-checker to be able to infer that `42` has type `Int8` etc. without the need for a type ascription. but when I remove them, I get a type error!
```
Expected (Integer → (?1 → ?2))
but got (Int8 → (Int8 → Int8))
Specifically, Integer doesn't match Int8
```

does this mean that you didn't make `#%integer-literal` polymorphic, you made `the` convert from an `Integer` to an `Int8` etc.?